### PR TITLE
Refactoring of LabelStore and LabelStoreRocksDB to improve testability

### PR DESCRIPTION
**Pull request by robwtelicent**
The main purpose of this PR is to improve the testability of `LabelStoreRocksDB` by abstracting the RocksDB functionality into the mockable `RocksDBHelper` class. This enables testing of label store functionality without the need to spin up RockDB and instead provide predictable behaviour through mocking.

As a side-effect of this work the extension of `AutoCloseable` has been moved from `LabelStoreRocksDB` to `LabelStore`. This enables the creation of all `LabelStore` instances to be wrapped in a `try-with-resources` statement ensuring the underlying database is always closed (where necessary).

### CODE REVIEW

File: rdf-abac-core/src/main/java/io/telicent/jena/abac/labels/RocksDBHelper.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static io.telicent.jena.abac.labels.Labels.LOG;
+
+public class RocksDBHelper {
+
+    private final static String CREATE_MESSAGE = "create of RocksDB label store";
```
Comment (TelicentPaul) [inline rdf-abac-core/src/main/java/io/telicent/jena/abac/labels/RocksDBHelper.java:13]: Utterly pedantic and I know predates your work.
```suggestion
    private final static String CREATE_MESSAGE = "creation of RocksDB label store";
```

**Review by TelicentPaul — APPROVED** [review]
Looks good

**Review by rvesse — APPROVED** [review]
Always nice to make it easier to test corner cases and other error conditions which are hard to provoke in tests otherwise 💯

**Review by rvesse — APPROVED** [review]
